parsers/ELEXON.py

```diff
@@ -0,0 +1,202 @@
+
+    params['FromDate'] = target_datetime.replace(
+        hours=span[0]).format('YYYY-MM-DD')
+    params['ToDate'] = target_datetime.replace(
+        hours=span[1]).format('YYYY-MM-DD')
+
+    if 'ELEXON_TOKEN' not in os.environ:
+        raise Exception('No ELEXON_TOKEN found! Please add it to secrets.env!')
+    params['APIKey'] = os.environ['ELEXON_TOKEN']
+    return session.get(ELEXON_ENDPOINT.format(report), params=params)
+
+
+def query_exchange(session, target_datetime=None):
+    params = {
+        'ServiceType': 'csv'
+    }
+    response = query_ELEXON('INTERFUELHH', session, params, target_datetime)
+    response = response.content.decode()
+    return response
+
+
+def query_production(session, target_datetime=None):
+    params = {
+        'ServiceType': 'csv'
+    }
+    response = query_ELEXON('FUELINST', session, params, target_datetime)
+    response = response.content.decode()
+    return response
+
+
+def parse_exchange(zone_key1, zone_key2, csv_text, target_datetime=None,
+                   logger=logging.getLogger(__name__)):
+    if not csv_text:
+        return None
+
+    sorted_zone_keys = sorted([zone_key1, zone_key2])
+
+    exchange = '->'.join(sorted_zone_keys)
+    data_points = list()
+    lines = csv_text.split('\n')
+
+    # check field count in report is as expected
+    field_count = len(lines[1].split(','))
+    if field_count != EXPECTED_FIELDS['INTERFUELHH']:
+        logger.warning(
+            'Expected {} fields in INTERFUELHH report, got {}'.format(
+                EXPECTED_FIELDS['INTERFUELHH'], field_count
+            )
+        )
+
+    for line in lines[1:-1]:
+        fields = line.split(',')
+
+        # create the datetime
+        datetime = datetime_from_date_sp(fields[1], fields[2])
+
+        data = {
+            'sortedZoneKeys': exchange,
+            'datetime': datetime,
+            'source': 'bmreports.com'
+        }
+
+        # positive value implies import to GB
+        multiplier = -1 if 'GB' in sorted_zone_keys[0] else 1
+        data['netFlow'] = float(fields[EXCHANGES[exchange]]) * multiplier
+        data_points.append(data)
+
+    if not target_datetime:
+        return data_points[-1]
+    else:
+        return data_points
+
+
+def parse_production(csv_text, target_datetime=None,
+                     logger=logging.getLogger(__name__)):
+    if not csv_text:
+        return None
+
+    data_points = list()
+    lines = csv_text.split('\n')
+
+    # check field count in report is as expected
+    field_count = len(lines[1].split(','))
+    if field_count != EXPECTED_FIELDS['FUELINST']:
+        logger.warning(
+            'Expected {} fields in FUELINST report, got {}'.format(
+                EXPECTED_FIELDS['FUELINST'], field_count
+            )
+        )
+
+    for line in lines[1:-1]:
+        fields = line.split(',')
+
+        # create the datetime
+        date_obj = dt.datetime.strptime(fields[3], '%Y%m%d%H%M%S')
```

Comment:
Are we sure input time string is always formatted in UTC?

Reply:
![image](https://user-images.githubusercontent.com/6525086/39698346-6361bf3a-51ec-11e8-9eae-2c62f9f5f6f4.png)

Yes, see previous correspondence they sent me on this. All timestamps are in GMT, the only local times are settlement date - settlement period combinations.